recipes-kernel/linux/linux-raspberrypi_6.6.bbappend

```diff
@@ -0,0 +1,18 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+COMPATIBLE_MACHINE:raspberrypi5 = "(raspberrypi5)"
+
+DOMU_DTB_NAME = "${RPI_SOC_FAMILY}-${MACHINE}-domu"
+XEN_DTB_NAME = "${RPI_SOC_FAMILY}-${MACHINE}-xen"
+RPI_KERNEL_DEVICETREE:append = " \
+    broadcom/${DOMU_DTB_NAME}.dtb \
+    broadcom/${XEN_DTB_NAME}.dtb \
+"
+
+KERNEL_IMAGETYPES:append = " Image.gz"
+
+SRC_URI:append = " \
+    ${@bb.utils.contains("DISTRO_FEATURES", "xen", "file://xen-config.cfg", "", d)} \
+    file://${DOMU_DTB_NAME}.dts;subdir=git/arch/${ARCH}/boot/dts/broadcom \
+    file://${XEN_DTB_NAME}.dts;subdir=git/arch/${ARCH}/boot/dts/broadcom \
```

Comment:
Still not clear to me how do you build it without adding to the makefile. 
I can tell why it works: because, according to the yaml we have the following line in deploy:
       - "tmp/deploy/images/%{RPI_MACHINE}/%{XT_XEN_DTB_NAME}"
        - "tmp/deploy/images/%{RPI_MACHINE}/bcm2712-rpi-5-b.dtb"
So rpi5 using bcm2712-rpi-5-b.dtb to boot and all changes you have done in xen are not applied

Reply:
But sources don't have tmp/deploy/images/%{RPI_MACHINE}/%{XT_XEN_DTB_NAME}. This should be built some how.
I carefully checked. All changes are applied. Boot script can't changes on vanilla bcm2712-rpi-5-b.dtb. This will cause error messages bu I don't see any error message and result chosen node looks good. Also %{XT_XEN_DTB_NAME} provides any correction to vanilla dtb like delete some nodes and adds xen,passthrough property to another nodes. Those changes are present.

Reply:
I believe this is yocto magic. Check `kernel-devicetree.bbclass` for more info and note `RPI_KERNEL_DEVICETREE` variable.

Reply:
yep. already discussed it with @dsemenets .